File: c/point.c
#include <stdlib.h>
#include <math.h>

#include <point.h>

Point Point_Initialize(double x, double y)
{
  Point self;
  self.x = x;
  self.y = y;
  return self;
}

void Point_Destroy(Point *self)
{
}

const double Point_Distance(const Point *self)
{
  return sqrt(self->x * self->x + self->y * self->y);
}

File: lua/point.lua
Point = {
  new = function(self, obj)
    obj = obj or {}
    setmetatable(obj, self)
    self.__index = self
    return obj
  end,

  distance = function(self)
    return math.sqrt(self.x * self.x + self.y * self.y)
  end
}

